packages/loop-harness/tests/test_loop_git_ephemeral.py

```
@@ -0,0 +1,962 @@
+) -> None:
+    # Simulates worktree_manager.create_worktree() reusing a worktree left dirty by a previous,
+    # interrupted action (Fix-10, PR #256 review, High). Without this check the leftover content
+    # would silently seed the new ephemeral index and could ride along into a candidate commit.
+    Path(linked_worktree.worktree_path, "tracked.txt").write_text(
+        "leftover from a previous interrupted action, never committed\n", encoding="utf-8"
+    )
+    runtime_dir = (
+        linked_worktree.project_dir / ".claude" / "loop" / LOOP_ID / "docker-runtime" / ACTION_ID
+    )
+
+    with pytest.raises(git_ephemeral.EphemeralGitInfrastructureError, match="dirty|status"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)